src/auth/AuthRegistry.cc

```diff
@@ -189,79 +189,70 @@ void AuthRegistry::get_supported_methods(
-    // i am client
-    if (methods) {
-      *methods = client_methods;
-    }
-    if (modes) {
-      switch (peer_type) {
-      case CEPH_ENTITY_TYPE_MON:
-      case CEPH_ENTITY_TYPE_MGR:
-	*modes = mon_client_modes;
-	break;
-      default:
-	*modes = client_modes;
-      }
-    }
-    return;
-  case CEPH_ENTITY_TYPE_MON:
-  case CEPH_ENTITY_TYPE_MGR:
-    // i am mon/mgr
-    switch (peer_type) {
-    case CEPH_ENTITY_TYPE_MON:
-    case CEPH_ENTITY_TYPE_MGR:
-      // they are mon/mgr
-      if (methods) {
-	*methods = cluster_methods;
-      }
-      if (modes) {
-	*modes = mon_cluster_modes;
-      }
+
+  if (modes) {
+    switch (get_conn_type(peer_type)) {
+    case CONN_TYPE_CLIENT_MON:
+      *modes = mon_client_modes;
+      break;
+    case CONN_TYPE_CLIENT:
+      *modes = client_modes;
+      break;
+    case CONN_TYPE_CLUSTER_MON:
+      *modes = mon_cluster_modes;
+      break;
+    case CONN_TYPE_CLUSTER:
+      *modes = cluster_modes;
+      break;
+    case CONN_TYPE_SERVICE_MON:
+      *modes = mon_service_modes;
+      break;
+    case CONN_TYPE_SERVICE:
+      *modes = service_modes;
       break;
     default:
-      // they are anything but mons
-      if (methods) {
-	*methods = service_methods;
-      }
-      if (modes) {
-	*modes = mon_service_modes;
-      }
+      ceph_abort();
     }
-    return;
-  default:
-    // i am a non-mon daemon
-    switch (peer_type) {
+  }
+
+  if (methods) {
+    switch (cct->get_module_type()) {
+    case CEPH_ENTITY_TYPE_CLIENT:
+      // i am client
+      *methods = client_methods;
+      return;
     case CEPH_ENTITY_TYPE_MON:
     case CEPH_ENTITY_TYPE_MGR:
-      // they are a mon daemon
-      if (methods) {
-	*methods = cluster_methods;
-      }
-      if (modes) {
-	*modes = mon_cluster_modes;
-      }
-      break;
-    case CEPH_ENTITY_TYPE_MDS:
-    case CEPH_ENTITY_TYPE_OSD:
-      // they are another daemon
-      if (methods) {
-	*methods = cluster_methods;
-      }
-      if (modes) {
-	*modes = cluster_modes;
+      // i am mon/mgr
+      switch (peer_type) {
+      case CEPH_ENTITY_TYPE_MON:
+      case CEPH_ENTITY_TYPE_MGR:
+        // they are mon/mgr
+        *methods = cluster_methods;
+        break;
+      default:
+        // they are anything but mons
+        *methods = service_methods;
       }
-      break;
+      return;
     default:
-      // they are a client
-      if (methods) {
-	*methods = service_methods;
-      }
-      if (modes) {
-	*modes = service_modes;
+      // i am a non-mon daemon
+      switch (peer_type) {
+      case CEPH_ENTITY_TYPE_MON:
+      case CEPH_ENTITY_TYPE_MGR:
+        // they are a mon daemon
+        *methods = cluster_methods;
+        break;
+      case CEPH_ENTITY_TYPE_MDS:
+      case CEPH_ENTITY_TYPE_OSD:
+        // they are another daemon
+        *methods = cluster_methods;
+        break;
+      default:
+        // they are a client
+        *methods = service_methods;
+        break;
```

Comment:
a strange combination of 'return's on two cases, and implicit break on the 'default'

Reply:
I think a least one of the `returns` and the `break` is from original code. I will clean them up.